### DISCUSSION TITLE
get_used_amount() and get_max_amout() from Bucket always return 0 ?

### DISCUSSION THREAD

**Discussion by nmathey** [MadeInPierre/finalynx#126] [Q&A] [answered]
Hello,

Je cherche à ajuster le target_amount d'un shareholder par rapport à l'usage d'un bucket précédemment utilisé.

Mon exemple perso: pour mon matelas de sécurité, je souhaite d'abord que soit pris entièrement mon bucket fonds euro et ensuite seulement piocher le reliquat nécessaire sur mon bucket Livret A pour atteindre le target de ce folder.

Je pensais donc faire cela:

```
Folder(
                "Emergency founds",
                children=[
                    SharedFolder(
                        "Fonds Euro",
                        bucket=bucket_euro,
                    ),
                    SharedFolder(
                        "Garanted",
                        bucket=bucket_garanti,
                        target_amount=12000-bucket_euro.get_max_amount(),
                    ),
                ],
                target=TargetMax(12000),
            ),
```

Rendu- KO:  le get_max_amount() semble renvoyer 0 donc le target_amount reste à 12000:
<img width="480" alt="Capture d’écran 2023-06-04 à 00 03 05" src="https://github.com/MadeInPierre/finalynx/assets/20896232/a32040f5-9400-43f4-bcff-b9ea0180bca7">

Si tu confirmes que ce n'est pas normal je fais issue :)

**Comment by MadeInPierre** [comment, marked as answer]
Hello, au moment de créer ton portefeuille Finalynx n'a pas encore récupéré les données de Finary (ou du cache), du coup toutes les lignes sont forcément à zéro :/ 

Serait-il possible de mettre les fonds euro ET les livrets dans le même bucket ? Chaque shared folder pioche dans l'ordre défini dans le bucket, donc ça devrait revenir au même (j'espère) ? 

S'il y a de l'utilité à séparer les buckets je veux bien ton objectif pour voir si on peut améliorer quelque chose :) J'ai déjà plusieurs limitations en tête à cause des buckets qui sont très compliquées à résoudre, c'est un peu la plus grosse galère dans finalynx...

**Reply by nmathey** [reply]
> au moment de créer ton portefeuille Finalynx n'a pas encore récupéré les données de Finary (ou du cache), du coup toutes les lignes sont forcément à zéro :/

Oui je viens de m'en rendre compte :)

J'ai donc refait un bucket unique Livret A + fonds euro en faisant attention à l'ordre des Lines à l'intérieur puis à l'ordre des mes folders dans mon portfolio: ca passe ! :)

<img width="392" alt="Capture d’écran 2023-06-04 à 00 25 42" src="https://github.com/MadeInPierre/finalynx/assets/20896232/1f80ec25-f21d-4b52-a85b-cb5f7ae17c7c">
